### PR TITLE
Make middlewares customizable

### PR DESCRIPTION
**Pull request by yuters**
This allows to easily extend the package middlewares to implement custom logic and exceptions.

### CODE REVIEW

**Review by luceos — CHANGES_REQUESTED** [review]
Relates to #575 

@yuters thanks for this, that's really nifty. One request though; can you allow (and document in the config) that setting the config key to `null` will disable them? That would allow for the most flexibility.

**Comment by yuters** [comment]
I was also thinking about the config not being an associative array, just a simple array of middlewares to prepend to the stack, so you can just remove the ones you don't need, or add to it. Do you think it's preferable?

**Comment by luceos** [comment]
@yuters that crossed my mind too, but I didn't want to put too much pressure on your first contribution 👍 

So please take that path, it's possibly less overhead in terms of necessary code as well.

File: assets/configs/tenancy.php [hunk 1]
```diff
@@ -30,6 +30,17 @@
         // Must implement \Hyn\Tenancy\Contracts\Website
         'website' => \Hyn\Tenancy\Models\Website::class
     ],
+    /**
+     * The package middlewares. Removing a middleware here will disable it.
+     * You can of course extend/replace them or add your own.
+     */
+    'middlewares' => [
```
Comment (fletch3555) [inline assets/configs/tenancy.php]: One comment.  `middlewares` sounds weird to me (native-born English speaker).  Much like `deer`, `fish`, and even `software`, the pluralization of `middleware` is often just `middleware`.  Also, for reference: https://english.stackexchange.com/questions/257120/middleware-vs-middlewares

Comment (yuters) [reply assets/configs/tenancy.php]: That’s a good point. I’m not a native English speaker.

What would you suggest to do in a foreach statement (ex. right now it’s foreach (middlewares as middleware)) to clearly distinguish the plural from the singular?

Comment (fletch3555) [reply assets/configs/tenancy.php]: I was thinking about that earlier. I personally tend to use shorthand inside loops, so in my own code I'd probably use something like `mw` or `item`. But that's not necessarily the best option in a package like this.  I'll leave it up to you to come up with something reasonable.  Don't worry too much about singular vs plural (though that is a very good practice in general)

**Comment by luceos** [comment]
Very happy with this, will see to a new tagged version asap!

**Pull request merged**